modxcms/revolution
# Support multi select in MODx.Browser when multiSelect: true is provided

### What does it do?

Allows the media browser to be instantiated with `multiSelect: true` to indicate the caller supports multiple file selections. Selecting multiple files is done with ctrl/shift  

By default, this is off, which preserves the current behavior of only allowing a single image to be inserted. 

(*Note: _technically_ the current 3.0.2 behavior is to allow multiple _selections_, but only the _first_ selected image is ever returned to the caller, so that's kinda broken)

For ease of implementation, the provided `onSelect` callback is simply called repeatedly for each of the selected files rather than introducing a new mechanism. 

### Why is it needed?

Enable implementations to allow multiple file selections in a single action. In this specific case, I'm looking at MoreGallery and Redactor, but anything else that uses MODx.Browser and can support multiple selections will benefit from this. 

### How to test

Importing images in MoreGallery 1.16.0-pl supports multiple selections and specifies `multiSelect: true` to enable it, so that can be used to test multiple files are being passed to the callback. (* this release will be available from Jan 11th.)

Alternatively, running something like this in the console can be used to confirm multiple callbacks are run:

```
MODx.load({
    xtype: 'modx-browser',
    closeAction: 'close',
    id: Ext.id(),
    multiple: true, /* indicates multiple browsers may be used on the page, without this it wont show up a second time */
    multiSelect: true, /* allow multiple selection inside the browser */
    onSelect: function(data) {
         console.log('received onSelect callback', data);
    },
}).show();
```

Also change `multiSelect: true` to `multiSelect: false` to see it support just the one selection. 

### Related issue(s)/PR(s)

Surprisingly enough, I haven't been able to find an issue for this even though it's been a common request for users of my extras over the years.
